File: java/christmas/src/test/java/christmas/MenuTest.java
package christmas;
import christmas.Model.Menu;
import org.junit.jupiter.api.Test;

import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

public class MenuTest {

    @Test
    void extractMenuNames() {
        Menu menu = new Menu();
        String[] menuNames = menu.extractMenuNames();
        assertNotNull(menuNames);
        assertTrue(menuNames.length > 0);
    }

    @Test
    void findCategory() {
        Menu menu = new Menu();
        String category = menu.findCategory("티본스테이크");
        assertNotNull(category);
        assertEquals("메인", category);

        // Test for a menu that doesn't exist
        String nonExistingCategory = menu.findCategory("NonExistingMenu");
        assertNull(nonExistingCategory);
    }

    @Test
    void getMenuPrice() {
        Menu menu = new Menu();
        int price = menu.getMenuPrice("아이스크림");
        assertEquals(5000, price);

        // Test for a menu that doesn't exist
        int nonExistingPrice = menu.getMenuPrice("NonExistingMenu");
        assertEquals(0, nonExistingPrice);
    }

    @Test
    void isInCategory() {
        Menu menu = new Menu();
        assertTrue(menu.isInCategory("아이스크림", "디저트"));
        assertFalse(menu.isInCategory("티본스테이크", "디저트"));
    }

    @Test
    void getMenuNamesByCategory() {
        Menu menu = new Menu();
        List<String> dessertMenuNames = menu.getMenuNamesByCategory("디저트");
        assertNotNull(dessertMenuNames);
        assertTrue(dessertMenuNames.size() > 0);
    }
}

File: kotlin/christmas/src/test/kotlin/christmas/MenuTest.kt
package christmas
import christmas.model.menu
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.Assertions.*

class MenuTest {

    @Test
    fun `extractMenuNames should return all menu names`() {
        val menu = menu()
        val menuNames = menu.extractMenuNames()

        assertNotNull(menuNames)
        assertTrue(menuNames.isNotEmpty())
    }

    @Test
    fun `findCategory existing menu should return category`() {
        val menu = menu()
        val category = menu.findCategory("티본스테이크")

        assertNotNull(category)
        assertEquals("메인", category)
    }

    @Test
    fun `findCategory non-existing menu should return null`() {
        val menu = menu()
        val category = menu.findCategory("치킨")

        assertNull(category)
    }

    @Test
    fun `getMenuPrice existing menu should return price`() {
        val menu = menu()
        val price = menu.getMenuPrice("제로콜라")

        assertTrue(price > 0)
    }

    @Test
    fun `getMenuPrice non-existing menu should return zero`() {
        val menu = menu()
        val price = menu.getMenuPrice("햄버거")

        assertEquals(0, price)
    }

    @Test
    fun `isInCategory menu in category should return true`() {
        val menu = menu()
        val isInCategory = menu.isInCategory("초코케이크", "디저트")

        assertTrue(isInCategory)
    }

    @Test
    fun `isInCategory menu not in category should return false`() {
        val menu = menu()
        val isInCategory = menu.isInCategory("레드와인", "메인")

        assertFalse(isInCategory)
    }

    @Test
    fun `getMenuNamesByCategory existing category should return menu names`() {
        val menu = menu()
        val category = "애피타이저"
        val menuNames = menu.getMenuNamesByCategory(category)

        assertNotNull(menuNames)
        assertTrue(menuNames.isNotEmpty())
    }

    @Test
    fun `getMenuNamesByCategory non-existing category should return empty list`() {
        val menu = menu()
        val category = "스낵"
        val menuNames = menu.getMenuNamesByCategory(category)

        assertNotNull(menuNames)
        assertTrue(menuNames.isEmpty())
    }
}
